profiles/lib/guidance.lua

```diff
@@ -1,154 +1,146 @@
+  end
+
+  local lane_count = input_way:get_value_by_key("lanes")
+  if lane_count then
+    local lc = tonumber(lane_count)
+    if lc ~= nil then
+      result.road_classification.num_lanes = lc
     end
-    if link_types[highway] then
-        result.road_classification.link_class = true;
-    end
-    if highway_classes[highway] ~= nil then
-        result.road_classification.road_priority_class = highway_classes[highway]
-    else
-        result.road_classification.road_priority_class = default_highway_class
+  else
+    local total_count = 0
+    local forward_count = input_way:get_value_by_key("lanes:forward")
+    if forward_count then
+      local fc = tonumber(forward_count)
+      if fc ~= nil then
+        total_count = fc
+      end
     end
-    if road_types[highway] then
-        result.road_classification.may_be_ignored = false;
-    else
-        result.road_classification.may_be_ignored = true;
+    local backward_count = input_way:get_value_by_key("lanes:backward")
+    if backward_count then
+      local bc = tonumber(backward_count)
+      if bc ~= nil then
+        total_count = total_count + bc
+      end
     end
-
-    local lane_count = input_way:get_value_by_key("lanes")
-    if lane_count then
-        local lc = tonumber(lane_count)
-        if lc ~= nil then
-            result.road_classification.num_lanes = lc
-        end
-    else
-        local total_count = 0
-        local forward_count = input_way:get_value_by_key("lanes:forward")
-        if forward_count then
-            local fc = tonumber(forward_count)
-            if fc ~= nil then
-                total_count = fc
-            end
-        end
-        local backward_count = input_way:get_value_by_key("lanes:backward")
-        if backward_count then
-            local bc = tonumber(backward_count)
-            if bc ~= nil then
-                total_count = total_count + bc
-            end
-        end
-        if total_count ~= 0 then
-            result.road_classification.num_lanes = total_count
-        end
+    if total_count ~= 0 then
+      result.road_classification.num_lanes = total_count
     end
+  end
 end
 
 -- returns forward,backward psv lane count
 local function get_psv_counts(way)
-    local psv = way:get_value_by_key("lanes:psv")
-    local psv_forward = way:get_value_by_key("lanes:psv:forward");
-    local psv_backward = way:get_value_by_key("lanes:psv:backward");
-
-    local fw = 0;
-    local bw = 0;
-    if  psv then
-        fw = tonumber(psv)
-        if( fw == nil ) then
-            fw = 0
-        end
-    end
-    if psv_forward then
-        fw = tonumber(psv_forward)
-        if( fw == nil ) then
-            fw = 0
-        end
-    end
-    if psv_backward then
-        bw = tonumber(psv_backward);
-        if( bw == nil ) then
-            bw = 0
-        end
-    end
-    return fw, bw
+  local psv_forward, psv_backward = Directional.get_values_by_key(way,'lanes:psv')
```

Comment:
to clarify, Directional.get_values_by_key() will first look for e.g. 'maxspeed:forward'. If that's not present it will loook for 'maxspeed'. This behaviour is correct for things like maxspeed, surface, access, etc.
Are there cases where it does not make sense? What about lanes?